graphbolt/src/cuda/index_select_impl.cu

```diff
@@ -0,0 +1,87 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ * @file cuda/index_select_impl.cu
+ * @brief Index select operator implementation on CUDA.
+ */
+#include <torch/script.h>
+
+#include <numeric>
+
+#include "../index_select.h"
+#include "./macro.h"
+
+namespace graphbolt {
+namespace ops {
+
+template <typename DType, typename IdType>
+__global__ void IndexSelectMultiKernel(
+    const DType* const input, const int64_t input_len,
+    const int64_t feature_size, const IdType* const index,
+    const int64_t output_len, DType* const output) {
+  int64_t out_row_index = blockIdx.x * blockDim.y + threadIdx.y;
+
+  const int64_t stride = blockDim.y * gridDim.x;
+
+  while (out_row_index < output_len) {
+    int64_t column = threadIdx.x;
+    const int64_t in_row = index[out_row_index];
+    assert(in_row >= 0 && in_row < input_len);
+    while (column < feature_size) {
+      output[out_row_index * feature_size + column] =
+          input[in_row * feature_size + column];
+      column += blockDim.x;
+    }
+    out_row_index += stride;
+  }
+}
+
+template <typename DType, typename IdType>
+torch::Tensor UVAIndexSelectImpl_(torch::Tensor input, torch::Tensor index) {
```

Comment:
Why not copy the `IndexSelectCPUFromGPU` from [here](https://github.com/dmlc/dgl/blob/7439b7e73bdb85b4285ab01f704ac5a4f77c927e/src/array/cuda/uvm/array_index_select_uvm.cu#L19)? There're many optimizations, such as the `IndexSelectMultiKernelAligned` kernel, sorting indices, etc.

Reply:
Yes, I will enable those optimizations later. But this PR will only add a single GPU kernel to demonstrate how to integrate GPU code in graphbolt.

Reply:
That makes sense.